airflow/hooks/fs_hooks/base.py

```diff
@@ -0,0 +1,231 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+from builtins import super
+import errno
+import importlib
+import posixpath
+
+from airflow.hooks.base_hook import BaseHook
+
+from . import _fnmatch as fnmatch
+
+_FS_BASE_MODULE = '.'.join(__name__.split('.')[:-1])
+
+
+class FsHook(BaseHook):
+    """Base FsHook defining the FsHook interface and providing some basic
+       functionality built on this interface.
+    """
+
+    _conn_classes = {
+        'ftp': _FS_BASE_MODULE + '.ftp.FtpHook',
+        'hdfs': _FS_BASE_MODULE + '.hdfs3.Hdfs3Hook',
+        'local': _FS_BASE_MODULE + '.local.LocalFsHook',
```

Comment:
Local FS is not good practice and shouldn't be included -- we don't want people to get in to the habit of storing files on disk IMO